lib/paper_trail/version_concern.rb

```diff
@@ -417,7 +417,14 @@ def reify_has_many_through(associations, model, options = {})
       associations.each do |assoc|
         next unless assoc.klass.paper_trail_enabled_for_model?
         through_collection = model.send(assoc.options[:through])
-        collection_keys = through_collection.map { |through_model| through_model.send(assoc.foreign_key) }
+
+        # if the association is a has_many association again, then call reify_has_manys for each through_collection
+        if !assoc.source_reflection.belongs_to? && through_collection.present?
+          through_collection.each { |through_model| reify_has_manys(through_model,options) }
+          return
+        end
+
+        collection_keys = through_collection.map { |through_model| through_model.send(assoc.association_foreign_key)}
```

Comment:
If I revert the change to this one line, by calling `assoc.foreign_key` instead of `assoc.association_foreign_key`, all the tests still pass.  So, is it necessary to change this line?